service/azureconfig/v1/resource/vnetpeering/update_test.go

```diff
@@ -0,0 +1,140 @@
+					ProvisioningState: to.StringPtr("some provisioning state"),
+				},
+			},
+			network.VirtualNetworkPeering{
+				Name: to.StringPtr("some Name"),
+				VirtualNetworkPeeringPropertiesFormat: &network.VirtualNetworkPeeringPropertiesFormat{
+					AllowVirtualNetworkAccess: to.BoolPtr(true),
+					RemoteVirtualNetwork: &network.SubResource{
+						ID: to.StringPtr("some ID"),
+					},
+				},
+			},
+			false,
+			false,
+		},
+		{
+			network.VirtualNetworkPeering{
+				// need an update when RemoteVirtualNetwork.ID property change
+				Name: to.StringPtr("some Name"),
+				VirtualNetworkPeeringPropertiesFormat: &network.VirtualNetworkPeeringPropertiesFormat{
+					AllowVirtualNetworkAccess: to.BoolPtr(true),
+					RemoteVirtualNetwork: &network.SubResource{
+						ID: to.StringPtr("some other ID"),
+					},
+				},
+			},
+			network.VirtualNetworkPeering{
+				Name: to.StringPtr("some Name"),
+				VirtualNetworkPeeringPropertiesFormat: &network.VirtualNetworkPeeringPropertiesFormat{
+					AllowVirtualNetworkAccess: to.BoolPtr(true),
+					RemoteVirtualNetwork: &network.SubResource{
+						ID: to.StringPtr("some ID"),
+					},
+				},
+			},
+			true,
+			false,
+		},
+		{
+			network.VirtualNetworkPeering{
+				// need an update when AllowVirtualNetworkAccess property change
+				Name: to.StringPtr("some Name"),
+				VirtualNetworkPeeringPropertiesFormat: &network.VirtualNetworkPeeringPropertiesFormat{
+					AllowVirtualNetworkAccess: to.BoolPtr(false),
+					RemoteVirtualNetwork: &network.SubResource{
+						ID: to.StringPtr("some ID"),
+					},
+				},
+			},
+			network.VirtualNetworkPeering{
+				Name: to.StringPtr("some Name"),
+				VirtualNetworkPeeringPropertiesFormat: &network.VirtualNetworkPeeringPropertiesFormat{
+					AllowVirtualNetworkAccess: to.BoolPtr(true),
+					RemoteVirtualNetwork: &network.SubResource{
+						ID: to.StringPtr("some ID"),
+					},
+				},
+			},
+			true,
+			false,
+		},
+		{
+			network.VirtualNetworkPeering{
+				// need an update when Name property change
+				Name: to.StringPtr("some Name"),
+				VirtualNetworkPeeringPropertiesFormat: &network.VirtualNetworkPeeringPropertiesFormat{
+					AllowVirtualNetworkAccess: to.BoolPtr(true),
+					RemoteVirtualNetwork: &network.SubResource{
+						ID: to.StringPtr("some ID"),
+					},
+				},
+			},
+			network.VirtualNetworkPeering{
+				Name: to.StringPtr("some other Name"),
+				VirtualNetworkPeeringPropertiesFormat: &network.VirtualNetworkPeeringPropertiesFormat{
+					AllowVirtualNetworkAccess: to.BoolPtr(true),
+					RemoteVirtualNetwork: &network.SubResource{
+						ID: to.StringPtr("some ID"),
+					},
+				},
+			},
+			true,
+			false,
+		},
+	}
+
+	for _, c := range cases {
+		result, err := needUpdate(c.current, c.desired)
+		if c.isError {
+			if err == nil {
+				t.Errorf("expected error got '%t'", err)
+			}
+		} else {
+			if result != c.expected {
+				t.Errorf("expected '%t' got '%t'", c.expected, result)
+			}
+		}
+	}
```

Comment:
It would be good to create new tests on par with https://github.com/giantswarm/fmt/blob/master/go/general_style_guide.md#unit-tests